// @mantine/charts/cjs/PieChart/PieChart.cjs
'use client';
'use strict';

var React = require('react');
var recharts = require('recharts');
var core = require('@mantine/core');
var ChartTooltip = require('../ChartTooltip/ChartTooltip.cjs');
var PieChart_module = require('./PieChart.module.css.cjs');

const defaultProps = {
  withTooltip: false,
  withLabelsLine: true,
  paddingAngle: 0,
  size: 160,
  strokeWidth: 1,
  startAngle: 0,
  endAngle: 360,
  tooltipDataSource: "all",
  labelsPosition: "outside"
};
const varsResolver = core.createVarsResolver(
  (theme, { strokeColor, labelColor, withLabels, size, labelsPosition }) => ({
    root: {
      "--chart-stroke-color": strokeColor ? core.getThemeColor(strokeColor, theme) : void 0,
      "--chart-labels-color": labelColor ? core.getThemeColor(labelColor, theme) : void 0,
      "--chart-size": withLabels && labelsPosition === "outside" ? core.rem(size + 80) : core.rem(size)
    }
  })
);
const insideLabel = ({ cx, cy, midAngle, innerRadius, outerRadius, value }) => {
  const RADIAN = Math.PI / 180;
  const radius = innerRadius + (outerRadius - innerRadius) * 0.5;
  const x = cx + radius * Math.cos(-midAngle * RADIAN);
  const y = cy + radius * Math.sin(-midAngle * RADIAN);
  return /* @__PURE__ */ React.createElement(
    "text",
    {
      x,
      y,
      textAnchor: x > cx ? "start" : "end",
      dominantBaseline: "central",
      className: PieChart_module.label
    },
    value
  );
};
const PieChart = core.factory((_props, ref) => {
  const props = core.useProps("PieChart", defaultProps, _props);
  const {
    classNames,
    className,
    style,
    styles,
    unstyled,
    vars,
    data,
    withTooltip,
    tooltipAnimationDuration,
    tooltipProps,
    pieProps,
    paddingAngle,
    withLabels,
    withLabelsLine,
    size,
    strokeWidth,
    startAngle,
    endAngle,
    tooltipDataSource,
    children,
    pieChartProps,
    labelsPosition,
    valueFormatter,
    ...others
  } = props;
  const theme = core.useMantineTheme();
  const getStyles = core.useStyles({
    name: "PieChart",
    classes: PieChart_module,
    props,
    className,
    style,
    classNames,
    styles,
    unstyled,
    vars,
    varsResolver
  });
  const { resolvedClassNames, resolvedStyles } = core.useResolvedStylesApi({
    classNames,
    styles,
    props
  });
  const cells = data.map((item, index) => /* @__PURE__ */ React.createElement(
    recharts.Cell,
    {
      key: index,
      fill: core.getThemeColor(item.color, theme),
      stroke: "var(--chart-stroke-color, var(--mantine-color-body))",
      strokeWidth
    }
  ));
  return /* @__PURE__ */ React.createElement(core.Box, { ref, size, ...getStyles("root"), ...others }, /* @__PURE__ */ React.createElement(recharts.ResponsiveContainer, null, /* @__PURE__ */ React.createElement(recharts.PieChart, { ...pieChartProps }, /* @__PURE__ */ React.createElement(
    recharts.Pie,
    {
      data,
      innerRadius: 0,
      outerRadius: size / 2,
      dataKey: "value",
      isAnimationActive: false,
      paddingAngle,
      startAngle,
      endAngle,
      label: withLabels ? labelsPosition === "inside" ? insideLabel : {
        fill: "var(--chart-labels-color, var(--mantine-color-dimmed))",
        fontSize: 12,
        fontFamily: "var(--mantine-font-family)"
      } : false,
      labelLine: withLabelsLine && labelsPosition === "outside" ? {
        stroke: "var(--chart-label-color, var(--mantine-color-dimmed))",
        strokeWidth: 1
      } : false,
      ...pieProps
    },
    cells
  ), withTooltip && /* @__PURE__ */ React.createElement(
    recharts.Tooltip,
    {
      animationDuration: tooltipAnimationDuration,
      isAnimationActive: false,
      content: ({ payload }) => /* @__PURE__ */ React.createElement(
        ChartTooltip.ChartTooltip,
        {
          payload: data,
          classNames: resolvedClassNames,
          styles: resolvedStyles,
          type: "radial",
          segmentId: tooltipDataSource === "segment" ? payload?.[0]?.name : void 0,
          valueFormatter
        }
      ),
      ...tooltipProps
    }
  ), children)));
});
PieChart.displayName = "@mantine/charts/PieChart";
PieChart.classes = PieChart_module;

exports.PieChart = PieChart;
