// @mantine/charts/esm/ChartTooltip/ChartTooltip.mjs
'use client';
import React from 'react';
import { factory, useProps, useMantineTheme, useStyles, ColorSwatch, getThemeColor, Box } from '@mantine/core';
import { getSeriesLabels } from '../utils/get-series-labels/get-series-labels.mjs';
import classes from './ChartTooltip.module.css.mjs';

function getFilteredChartTooltipPayload(payload, segmentId) {
  const duplicatesFilter = payload.filter((item) => item.fill !== "none" || !item.color);
  if (!segmentId) {
    return duplicatesFilter;
  }
  return duplicatesFilter.filter((item) => item.name === segmentId);
}
function getData(item, type) {
  if (type === "radial") {
    return item.value;
  }
  return item.payload[item.dataKey];
}
const defaultProps = {
  type: "area"
};
const ChartTooltip = factory((_props, ref) => {
  const props = useProps("ChartTooltip", defaultProps, _props);
  const {
    classNames,
    className,
    style,
    styles,
    unstyled,
    vars,
    payload,
    label,
    unit,
    type,
    segmentId,
    mod,
    series,
    valueFormatter,
    ...others
  } = props;
  const theme = useMantineTheme();
  const getStyles = useStyles({
    name: "ChartTooltip",
    classes,
    props,
    className,
    style,
    classNames,
    styles,
    unstyled
  });
  if (!payload) {
    return null;
  }
  const filteredPayload = getFilteredChartTooltipPayload(payload, segmentId);
  const labels = getSeriesLabels(series);
  const items = filteredPayload.map((item) => /* @__PURE__ */ React.createElement("div", { key: item.name, ...getStyles("tooltipItem") }, /* @__PURE__ */ React.createElement("div", { ...getStyles("tooltipItemBody") }, /* @__PURE__ */ React.createElement(
    ColorSwatch,
    {
      color: getThemeColor(item.color, theme),
      size: 12,
      ...getStyles("tooltipItemColor"),
      withShadow: false
    }
  ), /* @__PURE__ */ React.createElement("div", { ...getStyles("tooltipItemName") }, labels[item.name] || item.name)), /* @__PURE__ */ React.createElement("div", { ...getStyles("tooltipItemData") }, typeof valueFormatter === "function" ? valueFormatter(getData(item, type)) : getData(item, type), unit)));
  return /* @__PURE__ */ React.createElement(Box, { ...getStyles("tooltip"), mod: [{ type }, mod], ref, ...others }, label && /* @__PURE__ */ React.createElement("div", { ...getStyles("tooltipLabel") }, label), /* @__PURE__ */ React.createElement("div", { ...getStyles("tooltipBody") }, items));
});
ChartTooltip.displayName = "@mantine/charts/ChartTooltip";

export { ChartTooltip, getFilteredChartTooltipPayload };
